File: backend/src/Domain/Settings/Entities/VehicleCategory.cs
using System;

namespace DetailingArsenal.Domain {
    public class VehicleCategory : Entity<VehicleCategory>, IUserEntity {
        public const int NameMaxLength = 32;
        public const int DescriptionMaxLength = 128;

        public Guid UserId { get; set; } = Guid.Empty;

        /// <summary>
        /// Display name.
        /// </summary>
        public string Name {
            get => name;
            set {
                if (value.Length > NameMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                name = value;
            }
        }

        /// <summary>
        /// Longer description explaining the vehicle category.
        /// </summary>
        public string? Description {
            get => description;
            set {
                if (value?.Length > DescriptionMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                description = value;
            }
        }

        private string name = null!;
        private string? description = null;
    }
}

File: app/src/modules/app/api/settings/entities/vehicle-category.ts
import { Entity } from '@/core';

/**
 * Category to group price and time durations for services.
 */
export class VehicleCategory extends Entity {
    public static NAME_MAX_LENGTH = 32;
    public static DESCRIPTION_MAX_LENGTH = 128;

    /**
     * Create a new vehicle category.
     * @param name The display name of the category.
     * @param description The text description.
     */
    constructor(public name: string, public description: string = '') {
        super();

        if (name.length > VehicleCategory.NAME_MAX_LENGTH) {
            throw new RangeError('name');
        }

        if (description.length > VehicleCategory.DESCRIPTION_MAX_LENGTH) {
            throw new RangeError('description');
        }
    }
}
